tests/components/diagnostics/__init__.py

```diff
@@ -1,28 +1,49 @@
 """Tests for the Diagnostics integration."""
 from http import HTTPStatus
+from typing import Any
 
+from aiohttp.test_utils import TestClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntry
 from homeassistant.setup import async_setup_component
 
+from tests.conftest import TestClientGenerator
+
 
-async def _get_diagnostics_for_config_entry(hass, hass_client, config_entry):
+async def _get_diagnostics_for_config_entry(
+    hass: HomeAssistant,
+    hass_client: TestClientGenerator,
+    config_entry: ConfigEntry,
+) -> dict[str, Any]:
     """Return the diagnostics config entry for the specified domain."""
     assert await async_setup_component(hass, "diagnostics", {})
 
-    client = await hass_client()
+    client: TestClient = await hass_client()
     response = await client.get(
         f"/api/diagnostics/config_entry/{config_entry.entry_id}"
     )
     assert response.status == HTTPStatus.OK
     return await response.json()
 
 
-async def get_diagnostics_for_config_entry(hass, hass_client, config_entry):
+async def get_diagnostics_for_config_entry(
+    hass: HomeAssistant,
+    hass_client: TestClientGenerator,
+    config_entry: ConfigEntry,
+) -> Any:
```

Comment:
```suggestion
) -> dict[str, Any]:
```

Reply:
Diagnostics returns do not have to be a dict (it usually is, but not always)
See https://github.com/home-assistant/core/blob/a0e18051c7decbf24c94e8f54f7297aac17d03c5/homeassistant/components/diagnostics/__init__.py#L73-L81

Reply:
> Diagnostics returns do not have to be a dict (it usually is, but doesn't have to be)

That seems incorrectly typed upstream

Reply:
See #64976

Reply:
That is not correct and not the original design/idea behind the diagnostics.

Even if we decide to drop the dict[str, Any] idea, it still can be better typed than `Any`.

Reply:
zwave_js was the only one that didn't return a `dict`
This was adjusted and merged in #85524
I've moved back to `dict[str, Any]` so we should be good again.

Note: there are a few integrations that are currently return `dict` instead of `dict[str, Any]` so I have created a separate PR to clean those up: #85525 and #85526

Reply:
Resolved in preliminary PR